src/native/libs/System.Security.Cryptography.Native.Apple/extra_libs.cmake

```diff
@@ -5,9 +5,11 @@ macro(append_extra_cryptography_apple_libs NativeLibsExtra)
 
     list(APPEND ${NativeLibsExtra} ${COREFOUNDATION_LIBRARY} ${SECURITY_LIBRARY})
 
-    if (CLR_CMAKE_TARGET_OSX)
+    if (CLR_CMAKE_TARGET_OSX OR CLR_CMAKE_TARGET_MACCATALYST OR CLR_CMAKE_TARGET_IOS OR CLR_CMAKE_TARGET_TVOS)
```

Comment:
According to https://github.com/dotnet/runtime/issues/91736, the minimum iOS / tvOS target is 12.2. But we are strongly linking against CryptoKit which is not available until 13.0. What is the behavior of this? Will an App on 12.2 fail to launch because it can't find CryptoKit?

Reply:
Are we strongly linking into CryptoKit though? With Clang set to a specific min OS version the compiler uses the availability attributes in C headers, and the "text-based stub" (.tbd) linker files to automatically produce weak links. I would expect Swift to behave similarly.

Reply:
That could be. I am recalling though that folks on macOS 10.14 got app startup problems though on macOS (see https://github.com/dotnet/runtime/issues/93167 for an example). If it were weakly linked, I would have expected the app to work (even if it was on an unsupported OS).

Even if it is weakly linked, I wonder if our shims too-eagerly try to do something with it. Again using the previous example, that person was trying to use MD5, but they got CryptoKit errors.

Reply:
AFAIK #93167 happened after the Clang switch for minimum version was already bumped to 10.15. That automatically turns the weak links into strong ones since the library is part of the minimum target.

Reply:
Just a refresher on how it works:
1. The system headers have annotations for each API which says on which platform+version it's available
2. C compiler takes the `-m[macosx/ios/...]-version-min` switch which specifies the platform+version you are targeting
3. If there's mismatch between the annotation, usage and the minimum version, the C compiler with emit a warning/error
4. Conditional `if (__builtin_available(...))` block can be used to check for specific platform/version and avoid the warning/error by telling the compiler that the code can only run on given platform. This emits either a no-op or runtime check depending on the intersection with specified minimum supported platform version. Alternatively, an availability attribute can be place on the calling method and thus move the check requirement up the call chain.
5. Linker has import libraries/frameworks specified as .tbd files that include the minimal platform version. Based on intersection with the `-m[macosx/ios/...]-version-min` switch it generates either weak or strong imports.

.NET equivalents:
1. `SupportedOSPlatform` attribute
2. `SupportedOSPlatformVersion` property and TFM in .csproj
3. Platform Compatibility Analyzer running inside Roslyn (part of .NET SDK) will emit a warning/error
4. `if (OperatingSystem.IsXXX[VersionAtLeast](...))`; runtime/IL linker may turn it into no-op or compile it as runtime check

Swift equivalents:
1. `@available` attribute
2. Same as Clang
3. Swift compiler will emit warning/error
4. `if #available(iOS 15, *)` or `guard #available(iOS 15, *)`
5. Same linker as with Clang (ld64 / ld-prime), same mechanism

Reply:
Yes I checked and the compiler is automatically weakly-linking against CryptoKit.